src/analyses/base.ml

```diff
@@ -2355,6 +2355,40 @@ struct
           | _ -> failwith ("non-integer argument in call to function "^f.vname)
         end
       in
+      let apply_better_trig fk (float_fun : ?asPreciseAsConcrete:bool -> ?notInf_notNaN:bool -> FD.t -> FD.t) x =
```

Comment:
Is the type annotation needed, it looks a bit clunky.

Reply:
Possibly, because the type inference of optional arguments is not principal: https://dev.realworldocaml.org/variables-and-functions.html#inference-of-labeled-and-optional-arguments.